Intro to clustering r&tidymodels (#121)
microsoft/workshop-library: full/intro-clustering-R-tidymodels/README.md

## Goals

Hello and welcome to this learning adventure! In this folder, you will find a Clustering Challenge Notebook. This is basically an autograding guided assessment notebook that will help you test your understanding in using R to create models that predict which category an item belongs to. So grab your wand and let the adventure begin!

| **Goal**                      | Description                                    |
| ----------------------------- | -----------------------------------------------|
| **What will you learn**       | How to create clustering models in R           |
| **What you'll need**          | [Visual Studio Code](https://code.visualstudio.com?WT.mc_id=academic-59300-cacaste), [Docker Desktop](https://www.docker.com/products/docker-desktop), [Remote Developer Extension](https://aka.ms/vscode-remote/download/extension) and [Git](https://git-scm.com/downloads)   |
| **Duration**                  | 1.5 to 2 Hours                                         |
| **Slides**                    | [Powerpoint](./slides.pptx)                                 |

## Video

Coming soon!

## Pre-Learning

> This workshop allows learners to use the skills learnt in the module [Introduction to clustering models by using R and tidymodels](https://docs.microsoft.com/en-us/learn/modules/introduction-clustering-models/?WT.mc_id=academic-59300-cacaste) to create their own clustering models. As such, learners are encouraged to go through the module beforehand so as to be conversant with some of the concepts covered in this workshop.

Reviewer: Added some suggested changes for clarity

```suggestion
This workshop allows learners to use the skills learnt in the module [Introduction to clustering models by using R and tidymodels](https://docs.microsoft.com/en-us/learn/modules/introduction-clustering-models/?WT.mc_id=academic-59300-cacaste) to create their clustering models. As such, learners are encouraged to go through the module beforehand to get used to the concepts covered in this workshop.
```

> This workshop is the forth of a series designed to get started with data science in R. So you may also want to have a look to the [first](../explore-analyze-data-with-R), [second](../intro-regression-R-tidymodels) and [third](../intro-classification-R-tidymodels) workshops of the series, dealing with exploratory data analysis and supervised learning.

Reviewer: ```suggestion
This is the fourth workshop of a series designed to get started with data science in R. You may want to have a look at the [first](../explore-analyze-data-with-R), [second](../intro-regression-R-tidymodels), and [third](../intro-classification-R-tidymodels) workshops of the series, dealing with exploratory data analysis and supervised learning.
```

## Prerequisites

To get you up and running and writing R code in no time, we have containarized this workshop such that you have a ready out of the box R coding environment.

### Setting up the development container

> A **development container** is a running [Docker](https://www.docker.com) container with a well-defined tool/runtime stack and its prerequisites. You can try out development containers with **[GitHub Codespaces](https://github.com/features/codespaces)** or **[Visual Studio Code Remote - Containers](https://aka.ms/vscode-remote/containers)**.

Reviewer: ```suggestion
A **development container** is a running [Docker](https://www.docker.com) container with a well-defined tool and runtime stack with its prerequisites. You can try out development containers with **[GitHub Codespaces](https://github.com/features/codespaces)** or **[Visual Studio Code Remote - Containers](https://aka.ms/vscode-remote/containers)**.
```

#### GitHub Codespaces
Follow these steps to open this workshop in a Codespace:
1. Click the Code drop-down menu and select the **Open with Codespaces** option.
2. Select **+ New codespace** at the bottom on the pane.

For more info, check out the [GitHub documentation](https://docs.github.com/en/free-pro-team@latest/github/developing-online-with-codespaces/creating-a-codespace#creating-a-codespace?WT.mc_id=academic-59300-cacaste).

#### Binder
This workshop is also available on Binder. To open the notebook in a Binder environment, just click the button below.

[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/carlotta94c/workshop-library/introToClusteringR%26Tidymodels/?labpath=%2Ffull%2Fintro-clustering-R-tidymodels%2Fsolution%2FChallenge-Clustering.ipynb)

#### Learn Sandbox

> You can go through this challenge also leveraging on the Learn Sandbox environment, provided by [Unit 7](https://docs.microsoft.com/en-us/learn/modules/introduction-clustering-models/7-challenge-clustering?WT.mc_id=academic-59300-cacaste) of the MS Learn module - Introduction to clustering models by using R and tidymodels. Just sign in with your Microsoft or GitHub account and click on **Activate sandbox** to start.

Reviewer: ```suggestion
You can go through this challenge leveraging on the Learn Sandbox environment, provided by [Unit 7](https://docs.microsoft.com/en-us/learn/modules/introduction-clustering-models/7-challenge-clustering?WT.mc_id=academic-59300-cacaste) of the MS Learn module - Introduction to clustering models by using R and tidymodels. Just sign in with your Microsoft or GitHub account and click on **Activate sandbox** to start.
```

#### VS Code Remote - Containers
Follow these steps to open this workshop in a container using the VS Code Remote - Containers extension:

1. If this is your first time using a development container, please ensure your system meets the pre-reqs (i.e. have Docker installed) in the [getting started steps](https://aka.ms/vscode-remote/containers/getting-started?WT.mc_id=academic-59300-cacaste).

2. Press <kbd>F1</kbd> select and **Add Development Container Configuration Files...** command for **Remote-Containers** or **Codespaces**.

   > **Note:** If needed, you can drag-and-drop the `.devcontainer` folder from this sub-folder in a locally cloned copy of this repository into the VS Code file explorer instead of using the command.

> 3. Select this definition. You may also need to select **Show All Definitions...** for it to appear.

Reviewer: What does this  mean by _"this definition"_ ? Which one is the definition I would need to select?
